components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt

```diff
@@ -369,13 +370,13 @@ abstract class AbstractFetchDownloadService : Service() {
 
         response.body.useStream { inStream ->
             val newDownloadState = download.withResponse(response.headers, inStream)
-            downloadJobs[download.id]?.state = newDownloadState
+            currentDownloadJobState.state = newDownloadState
 
             useFileStream(newDownloadState, isResumingDownload) { outStream ->
-                copyInChunks(downloadJobs[download.id]!!, inStream, outStream)
+                copyInChunks(currentDownloadJobState, inStream, outStream)
```

Comment:
I think this ‼️ could be the source of [this crash ](https://sentry.prod.mozaws.net/operations/fenix-nightly/issues/7253086/?query=is:unresolved%20AbstractFetchDownloadService). What I'm not completely sure is how we got it that state as don't see any place where we are removing thing from the 
```

kotlin.KotlinNullPointerException: null
    at mozilla.components.feature.downloads.AbstractFetchDownloadService.useFileStreamScopedStorage(AbstractFetchDownloadService.kt:18)
    at mozilla.components.feature.downloads.AbstractFetchDownloadService.useFileStream$feature_downloads_release(AbstractFetchDownloadService.kt:8)
    at mozilla.components.feature.downloads.AbstractFetchDownloadService$performDownload$1.invoke(AbstractFetchDownloadService.kt:20)
    at mozilla.components.concept.fetch.Response$Body.useStream(Response.kt:1)
    at mozilla.components.feature.downloads.AbstractFetchDownloadService.performDownload$feature_downloads_release(AbstractFetchDownloadService.kt:19)
    at mozilla.components.feature.downloads.AbstractFetchDownloadService.startDownloadJob$feature_downloads_release(AbstractFetchDownloadService.kt:3)
    at mozilla.components.feature.downloads.AbstractFetchDownloadService$onStartCommand$1.invokeSuspend(AbstractFetchDownloadService.kt:5)
    at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:2)
    at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:19)
    at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:1)
    at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:14)

```

Reply:
I think we should add a comment explaining why we made these changes and what we aren't sure of so people working on this in the future will have a trail to follow.

Reply:
👍